CONTRIBUTING.md

```diff
@@ -0,0 +1,107 @@
+PRs should generally address only 1 issue at a time. If you need to fix two bugs, open two separate PRs. This will keep the scope of your pull requests smaller and allow them to be reviewed and merged more quickly.
+
+Each PR needs to be linked to a Github issue in [rancher project](https://github.com/rancher/rancher) which provides as much detail as possible on the commit at its motivation. If it is a bug fix, please describe how to reproduce it in the issue. Make the team or Project Manager aware of the Github issue so that they can set the correct milestone. Milestones are very important for release and QA tracking.
+
+Generally, pull requests should consist of a single logical commit. However, if your PR is for a large feature, you may need a more logical breakdown of commits. This is fine as long as each commit is a single logical unit. Changes generated by `go mod tidy` or `go generate` should go in separate commits.
+
+Pull requests should target the release/vX.Y branch instead of master. The latest code in RKE is included in such release branches.
+
+### Writing good commit messages
+Git commit messages should explain the how and why of your change and be separated into a brief subject line followed by a more detailed body. When in doubt, follow this guide for good commit messages and you can’t go wrong: https://chris.beams.io/posts/git-commit/.
+
+One particularly useful point made in the above guide is regarding commit subject lines:
+
+> A properly formed Git commit subject line should always be able to complete the following sentence:
+> 
+> - If applied, this commit will <ins>your subject line here</ins>
+
+A simple but effective convention to follow for commits is the “problem / solution” pattern. It looks like this:
+```
+<Subject>
+
+Problem: <Statement of problem>
+
+Solution: <Statement of solution>
+```
+
+As an example, here is a commit taken from the rancher/rancher repo:
+```
+commit b71ce2892eecb7c87a5212e3486f1de899a694aa
+Author: Dan Ramich <danold215@gmail.com>
+Date:   Tue Jun 19 11:56:52 2018 -0700
+
+    Add Validator for RoleTemplate
+
+    Problem:
+    Builtin RoleTemplates can be updated through the API
+
+    Solution:
+    Add a Validator to ensure the only field that can be changed on a
+    builtin RoleTemplate is 'locked'
+```
+
+### Reviewing, addressing feedback, and merging
+Generally, pull requests need two approvals from maintainers to be merged.
+
+When addressing review feedback, it is helpful to the reviewer if additional changes are made in new commits. This allows the reviewer to easily see the delta between what they previously reviewed and the changes you added to address their feedback.
+
+Once a PR has at least two approvals, it can be merged. Here’s how the merge should be handled:
+- If the PR is a single logical commit, the merger should use the “Rebase and merge” option. This keeps the git commit history very clean and simple and eliminates noise from "merge commits."
```

Comment:
Are the two first points backwards? If a PR has a single commit with added/updated logic `Create a merge commit` and if 1+ logic commits then `Rebase and merge`. Will also wait for @superseb's comments to be addressed and re-review this section after that.

Reply:
Good catch! I am missing the part where the commits should be squashed together once the PR has been approved